pkgs/by-name/po/porn-vault/porn-vault.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+cd $(mktemp -d)
+for i in %out%/share/porn-vault/*; do
```

Comment:
Is it truly necessary to link the JS code here, could we not just do `node $out/...` via an absolute (Nix-store) path?

Reply:
As I mentioned [in my other comment](https://github.com/NixOS/nixpkgs/pull/355785#discussion_r1842102289), PV expects to be run on the same workdir as it is stored, so you can’t just do “node $out/…” without cding into it because it will crash. But we can’t just cd into store, because PV also creates a tmp folder on your current workdir. So if you run it cding into store, it will fail due to the read-only filesystem.

That’s the reason making a tmp folder, linking stuff on it and cding into it is needed. 

As I mentioned on the other comment, an alternative would be symlinking $out/share/porn-vault/tmp to /tmp/porn-vault, and then just cding into store. That works and personally I prefer this approach, but is it actually allowed to symlink to something mutable from inside the store?